src/Storefront/Resources/app/storefront/src/plugin-system/plugin.manager.js

```diff
@@ -266,18 +266,92 @@ class PluginManagerSingleton {
         }
 
         if (typeof selector === 'string') {
-            selector = document.querySelectorAll(selector);
+            selector = PluginManagerSingleton._queryElements(selector, pluginClass, options, pluginName);
+            if (!selector) {
+                return;
+            }
         }
 
         return Iterator.iterate(selector, el => {
             PluginManagerSingleton._initializePluginOnElement(el, pluginClass, options, pluginName);
         });
     }
 
+    /**
+     * Determs the way to query the elements.
+     *
+     * [data-*] => querySelectorAll
+     * #fooBar => getElementById
+     * #foo_bar => getElementById
+     * #foo-bar => getElementById
+     * #foo .bar => querySelectorAll
+     * .fooBar => getElementsByClassName
+     * .FOO => getElementsByClassName
+     * .foo_bar => getElementsByClassName
+     * .foo-bar => getElementsByClassName
+     * .foo .bar => querySelectorAll
+     *
+     * FOO => getElementsByTagName
+     * FOO .bar => querySelectorAll
+     *
+     * For performance reason used regex based on common characters `a-zA-Z1-9_-`
+     * instead of the entire compatible characters
+     *
+     * @param {string} selector
+     * @param {Plugin} pluginClass
+     * @param {Object} options
+     * @param {string} pluginName
+     *
+     * @return {null|NodeList|HTMLCollection}
+     */
+    static _queryElements(selector, pluginClass, options, pluginName) {
+        // While the use of attributes is the most common right now, we should return early.
+        if (selector.startsWith('[')) {
+            return document.querySelectorAll(selector);
+        }
+
+        if (selector.startsWith('.')) {
+            const regexEl = /^.([\w-]+)$/.exec(selector);
+            if (regexEl) {
+                return document.getElementsByClassName(regexEl[1]);
+            }
+        } else if (selector.startsWith('#')) {
+            const regexEl = /^#([\w-]+)$/.exec(selector);
+            if (regexEl) {
+                // When we have a valid id-selector, we have to register it directly on element instead of returning a list.
+                PluginManagerSingleton._initializePluginOnElementById(regexEl[1], pluginClass, options, pluginName);
+                return null;
```

Comment:
Instead of calling `_initializePluginOnElementById` by yourself and then returning `null`, you should return a collection with a single element and use the existing call to `_initializePluginOnElement`.

Reply:
Damn! I really like and I'm ashamed of it.